src/components/MainArea.tsx

```diff
@@ -1,20 +1,59 @@
-import { useState } from "react";
+import { useEffect, useLayoutEffect, useState } from "react";
 import { ShadowDom } from "./ShadowDom.tsx";
-import CreateRecipe from "./CreateRecipe.tsx";
-import Recipes from "./Recipes.tsx";
-import { Route, Routes } from "react-router-dom";
-import CheckRecipes from "./CheckRecipes.tsx";
+import CreateRecipe from "./recipes/CreateRecipe.tsx";
+import Recipes from "./recipes/Recipes.tsx";
+import { Route, Routes, useLocation, useNavigate } from "react-router-dom";
+import CheckRecipes from "./cart/CheckRecipes.tsx";
+import Products from "./products/Products.tsx";
 
 const MainArea = () => {
   const [parentElement] = useState(() =>
-    document.querySelector("#pageFullWidth"),
+    document.querySelector<HTMLElement>("#pageFullWidth"),
   );
 
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  useEffect(() => {
```

Comment:
Spolu s timhle bych tam dal i https://github.com/Mczella/plugin/pull/13/files#diff-fe902ecf8656a5064a714d9d299490fe148af01571b08138c92702d9fa4a39daR42 sice je to takovej patch.. ale dokud nevymyslime nic lepsiho..